mgmt/RecordsConfig.cc

```diff
@@ -1168,7 +1168,10 @@ static const RecordElement RecordsConfig[] =
   ,
   {RECT_CONFIG, "proxy.config.ssl.client.groups_list", RECD_STRING, nullptr, RECU_RESTART_TS, RR_NULL, RECC_NULL, nullptr, RECA_NULL}
   ,
-
+  {RECT_CONFIG, "proxy.config.ssl.server.max_early_data", RECD_INT, "0", RECU_RESTART_TS, RR_NULL, RECC_NULL, nullptr, RECA_NULL}
+  ,
+  {RECT_CONFIG, "proxy.config.ssl.server.recv_max_early_data", RECD_INT, "16384", RECU_RESTART_TS, RR_NULL, RECC_NULL, nullptr, RECA_NULL}
```

Comment:
This enables early data by default? That seems aggressive. Also, we should make this not require restart, it seems to be a per-connection setting, and as such, should be possible to have a callback that atomically swaps the setting (as we do most of the time).

Reply:
No, `proxy.config.ssl.server.max_early_data` enables early data if set to anything other than `0`, `proxy.config.ssl.server.recv_max_early_data` has a default value of `16384` defined by openssl: https://www.openssl.org/docs/manmaster/man3/SSL_CTX_set_recv_max_early_data.html

Reply:
Hmmm, then what's the difference between the two ? What value do I set proxy.config.ssl.server.max_early_data to ?

Reply:
They're totally different settings, `proxy.config.ssl.server.max_early_data` controls whether the early data feature is enabled and if enabled how much it accepts, `proxy.config.ssl.server.recv_max_early_data` is more of a tolerance limit of how much early data the server can receive without aborting the connection.

https://github.com/mattcaswell/openssl/commit/f5c8276647bd88854f4df876558fa91c7053ba8e